### PR TITLE
Fix forbiddenapis on java 11 

### LINKED ISSUE #33082: [CI] Bundled signatures resource not found: jdk-unsafe-11 for java8/java10
Build fails with 

```
04:05:42 ERROR: IO problem while reading files with API signatures: java.io.FileNotFoundException: Bundled signatures resource not found: jdk-unsafe-11
04:05:42   No history is available.
04:05:42 Starting process 'command '/var/lib/jenkins/.java/java8/bin/java''. Working directory: /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/server Command: /var/lib/jenkins/.java/java8/bin/java -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp toooone_of_jars_here:/var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual && linux/server/build/classes/java/main:/var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual && linux/server/build/resources/main de.thetaphi.forbiddenapis.cli.CliMain -f /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/server/build/build-tools-exported/forbidden/jdk-signatures.txt -f /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/server/build/build-tools-exported/forbidden/es-all-signatures.txt -f /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/server/build/build-tools-exported/forbidden/es-server-signatures.txt --suppressannotation **.SuppressForbidden -b jdk-unsafe-11 -b jdk-deprecated-11 -b jdk-non-portable -b jdk-system-out -d /var/lib/jenkins/workspace/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA/java11/ES_RUNTIME_JAVA/java8/nodes/virtual&&linux/server/build/classes/java/main
04:05:42 Successfully started process 'command '/var/lib/jenkins/.java/java8/bin/java''
04:05:42 Scanning for classes to check...
04:05:42 :server:forbiddenApisMain (Thread[Task worker for ':' Thread 4,5,main]) completed. Took 0.434 secs.
04:05:42 49 actionable tasks: 17 executed, 32 up-to-date
04:05:42 
04:05:42 FAILURE: Build failed with an exception.
04:05:42 
04:05:42 * What went wrong:
04:05:42 Execution failed for task ':server:forbiddenApisMain'.
04:05:42 > Process 'command '/var/lib/jenkins/.java/java8/bin/java'' finished with non-zero exit value 4
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/254/console

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8,nodes=virtual&&linux/255/console

### PR DESCRIPTION
**Pull request by alpar-t**
Forbidden APIs doesn't yet support Java 11, so use the signatures from 10 when running on 11.

Closes #33082.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java [hunk 1]
```diff
@@ -41,6 +43,7 @@
 
 public class ForbiddenApisCliTask extends DefaultTask {
 
+    private final Logger logger = Logging.getLogger(ForbiddenApisCliTask.class);
```
Comment (vladimirdolzhenko) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ForbiddenApisCliTask.java]: should it be `static` ?
`private static final Logger logger`

File: x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/GetJobRequest.java [hunk 2]
```diff
@@ -90,7 +90,6 @@ public GetJobRequest(String... jobIds) {
 
     /**
      * See {@link GetJobRequest#isAllowNoJobs()}
-     * @param allowNoJobs
```
Comment (vladimirdolzhenko) [inline x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/GetJobRequest.java]: don't need to drop the javadoc param

**Review by vladimirdolzhenko — CHANGES_REQUESTED** [review]
look good, I left a minor note

**Review by vladimirdolzhenko — APPROVED** [review]

File: x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/GetJobRequest.java (hunk 2 above)
Comment (vladimirdolzhenko) [reply x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/GetJobRequest.java]: surprise, surprise

**Comment by vladimirdolzhenko** [comment]
LGTM

**Comment by nik9000** [comment]
We didn't used to make loggers static because we couldn't get the node name
in them. I fixed that recenlty. I prefer static but don't care enough to
block any PRs over it. And there are still some cars where it can't be
static, namely when they include the shard or index id. Long story short:
we did the old thing for a reason that is no longer true, use your best
judgement now and don't rely on past patterns.

**Pull request merged**